### PR TITLE
Use checked methods from `ColumnFamily`

### LINKED ISSUE #8787: Preconditions for Create, Update and Delete in `ColumnFamily`
Modifications to the state go through the `ColumnFamily` interface which, among other methods, supports `put` and `delete` We can expand `put` to `create` and `update` and do something similar for `delete`.

### PR DESCRIPTION
**Pull request by lenaschoenburg**
## Description

Replaces all usages of `ColumnFamily#put` with `insert`, `update` or `upsert`  and  `delete` with `deleteExisting` or `deleteIfExists`

I've tried to not use `upsert` where possible because this method doesn't check anything and is essentially just a renamed `put`.  To get rid of all of them would require more refactoring than I'd like in this PR

## Related issues

closes #8787

### CODE REVIEW

**Comment by pihme** [comment]
> Replaces all usages of `ColumnFamily#put` with `insert`, `update` or `upsert`. The `deleteExisting` method is not yet used, mostly because the benefit over risk ratio is a bit worse.

I do see some usage of `deleteExistng`though. Please explain.

 
> This PR also adds a config value to disable the checks.

Is this still accurate?

**Comment by oleschoenburg** [comment]
The PR description was outdated, I'll update.

File: engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbEventScopeInstanceState.java [hunk 1]
```diff
@@ -53,7 +53,7 @@ public void shutdownInstance(final long eventScopeKey) {
     if (instance != null) {
       this.eventScopeKey.wrapLong(eventScopeKey);
       instance.setAccepting(false);
-      eventScopeInstanceColumnFamily.put(this.eventScopeKey, instance);
+      eventScopeInstanceColumnFamily.update(this.eventScopeKey, instance);
```
Comment (pihme) [inline engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbEventScopeInstanceState.java:56]: @korthout Do we still need this method `shotdownInstance`? It is only used in tests. It also doesn't seem to be a helper method, but a method that was used in the past

File: engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java [hunk 2]
```diff
@@ -144,7 +144,7 @@ public void timeout(final long key, final JobRecord record) {
     validateParameters(type);
     EnsureUtil.ensureGreaterThan("deadline", deadline, 0);
 
-    createJob(key, record, type);
+    updateJob(key, record, State.ACTIVATABLE);
```
Comment (pihme) [inline engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java:147]: :question: Please explain this change. This seems to be different than the others and it is not clear to me, why this is a valid change or even a necessary one.

File: engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java [hunk 3]
```diff
@@ -376,11 +391,11 @@ private void makeJobNotActivatable(final DirectBuffer type) {
...
 
     jobTypeKey.wrapBuffer(type);
-    activatableColumnFamily.delete(typeJobKey);
+    activatableColumnFamily.deleteIfExists(typeJobKey);
   }
 
   private void removeJobDeadline(final long deadline) {
     deadlineKey.wrapLong(deadline);
-    deadlinesColumnFamily.delete(deadlineJobKey);
+    deadlinesColumnFamily.deleteIfExists(deadlineJobKey);
```
Comment (pihme) [inline engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java:399]: :x: Pretty sure this should be a _deleteExisting_. Please try run tests, etc.

File: engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java [hunk 4]
```diff
@@ -217,8 +217,8 @@ public JobRecord updateJobRetries(final long jobKey, final int retries) {
   }
 
   private void createJob(final long key, final JobRecord record, final DirectBuffer type) {
-    resetVariablesAndUpdateJobRecord(key, record);
-    updateJobState(State.ACTIVATABLE);
+    createJobRecord(key, record);
```
Comment (pihme) [inline engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java:220]: :question: Please explain this change. It is not clear to be why this change was needed and if it's equivalent behaviour

File: engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java [hunk 5]
```diff
@@ -348,16 +349,28 @@ private void notifyJobAvailable(final DirectBuffer jobType) {
     }
   }
 
+  private void createJobRecord(final long key, final JobRecord record) {
```
Comment (pihme) [inline engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java:352]: :question: Please also explain this change

File: engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java [hunk 6]
```diff
@@ -348,16 +349,28 @@ private void notifyJobAvailable(final DirectBuffer jobType) {
...
+
   private void resetVariablesAndUpdateJobRecord(final long key, final JobRecord updatedValue) {
     jobKey.wrapLong(key);
     // do not persist variables in job state
     jobRecordToWrite.setRecordWithoutVariables(updatedValue);
-    jobsColumnFamily.put(jobKey, jobRecordToWrite);
+    jobsColumnFamily.update(jobKey, jobRecordToWrite);
+  }
+
+  private void initializeJobState() {
```
Comment (pihme) [inline engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java:366]: :question: Please also explain this change

**Review by pihme — CHANGES_REQUESTED** [review]
First batch of review. Need to take a break. Reviewed until `DbJobState

@oleschoenburg Please let me know if I should continue with the review to the end, or you want to address the first batch of comments first

**Comment by pihme** [comment]
@korthout I need to call in the cavalry here. Could you please review the changes to:
* DbElementInstanceState
* DBJobState
* (more to come as I complete the review)

And look at comments where I tagged you. 

I would also appreciate if if you could be the third pair of eyes on this. But be warned, it's a tricky review :-)

File: engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbDeploymentState.java [hunk 7]
```diff
@@ -60,27 +60,27 @@ public DbDeploymentState(
...
-    pendingDeploymentColumnFamily.put(deploymentPartitionKey, DbNil.INSTANCE);
+    pendingDeploymentColumnFamily.insert(deploymentPartitionKey, DbNil.INSTANCE);
   }
 
   @Override
   public void removePendingDeploymentDistribution(final long deploymentKey, final int partition) {
     this.deploymentKey.wrapLong(deploymentKey);
     partitionKey.wrapInt(partition);
-    pendingDeploymentColumnFamily.delete(deploymentPartitionKey);
+    pendingDeploymentColumnFamily.deleteIfExists(deploymentPartitionKey);
```
[the remark this answers is not in the document]
Comment (oleschoenburg) [reply engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbDeploymentState.java:70]: Seems to work, except for the helpfully named `shouldRemovePendingDeploymentIdempotent` test that tests that removing a pending deployment twice works. I don't know _why_ that test exists though.

File: engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbProcessState.java [hunk 8]
```diff
@@ -133,12 +133,12 @@ public void putProcess(final long key, final ProcessRecord processRecord) {
   private void persistProcess(final long processDefinitionKey, final ProcessRecord processRecord) {
     persistedProcess.wrap(processRecord, processDefinitionKey);
     this.processDefinitionKey.wrapLong(processDefinitionKey);
-    processColumnFamily.put(this.processDefinitionKey, persistedProcess);
+    processColumnFamily.upsert(this.processDefinitionKey, persistedProcess);
```
[the remark this answers is not in the document]
Comment (oleschoenburg) [reply engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbProcessState.java:136]: Unfortunately not. There are tests that cover this, for example `shouldFilterDuplicateProcess` which deploys the same process twice.

File: engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java (hunk 3 above)
Comment (oleschoenburg) [reply engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java:399]: That doesn't work at all, lot's of failing tests. There doesn't seem to be a specific test for this but `shouldFailJobAndRetryWithBackOff` is one that fails.

File: broker/src/main/java/io/camunda/zeebe/broker/exporter/stream/ExportersState.java [hunk 9]
```diff
@@ -71,7 +71,7 @@ public long getLowestPosition() {
 
   public void removePosition(final String exporter) {
     exporterId.wrapString(exporter);
-    exporterPositionColumnFamily.delete(exporterId);
+    exporterPositionColumnFamily.deleteExisting(exporterId);
```
[the remark this answers is not in the document]
Comment (oleschoenburg) [reply broker/src/main/java/io/camunda/zeebe/broker/exporter/stream/ExportersState.java]: That makes sense to be, I've updated this to use `deleteIfExists` :+1:

File: engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbDecisionState.java [hunk 10]
```diff
@@ -134,11 +134,11 @@ public DbDecisionState(
   public void putDecision(final DecisionRecord record) {
     dbDecisionKey.wrapLong(record.getDecisionKey());
     dbPersistedDecision.wrap(record);
-    decisionsByKey.put(dbDecisionKey, dbPersistedDecision);
+    decisionsByKey.upsert(dbDecisionKey, dbPersistedDecision);
```
[the remark this answers is not in the document]
Comment (oleschoenburg) [reply engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbDecisionState.java]: You are right (but I could've sworn this didn't work when I first tried it :shrug:). I've updated to `insert` and adjusted the method name :+1:

File: engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbDecisionState.java [hunk 11]
```diff
@@ -134,11 +134,11 @@ public DbDecisionState(
...
 
     dbDecisionKey.wrapLong(record.getDecisionKey());
     dbDecisionRequirementsKey.wrapLong(record.getDecisionRequirementsKey());
-    decisionKeyByDecisionRequirementsKey.put(
+    decisionKeyByDecisionRequirementsKey.upsert(
```
[the remark this answers is not in the document]
Comment (oleschoenburg) [reply engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbDecisionState.java]: You are also right here, updated to `insert` and adjusted the method name.

File: engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbDecisionState.java [hunk 12]
```diff
@@ -148,7 +148,7 @@ public void putDecision(final DecisionRecord record) {
   public void putDecisionRequirements(final DecisionRequirementsRecord record) {
     dbDecisionRequirementsKey.wrapLong(record.getDecisionRequirementsKey());
     dbPersistedDecisionRequirements.wrap(record);
-    decisionRequirementsByKey.put(dbDecisionRequirementsKey, dbPersistedDecisionRequirements);
+    decisionRequirementsByKey.upsert(dbDecisionRequirementsKey, dbPersistedDecisionRequirements);
```
[the remark this answers is not in the document]
Comment (oleschoenburg) [reply engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbDecisionState.java]: Right again, you are on a roll :smile:. Updated to `insert` and adjusted the method name.

File: engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbDeploymentState.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (pihme) [reply engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbDeploymentState.java:70]: @korthout Any idea about this test and whether it is important?

**Comment by oleschoenburg** [comment]
Thanks for the review so far @pihme. I think it makes sense to continue like this. I've addressed most of your suggestions now and will spend some more time to explain the less obvious changes to `DbJobState` tomorrow.

File: engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java (hunk 4 above)
Comment (oleschoenburg) [reply engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java:220]: `createJobRecord` is just `resetVariablesAndUpdateJobRecord` except that it uses `insert`, not `update`.
`initializeJobState` is just `updateJobState` except that it uses `insert`, not `update`. 

Essentially I've introduced new methods so that we can split the insert and the update code paths.
It took me a while to understand this again though, so maybe the changes are either not worth it or we could do some renaming of methods here. WDYT?

File: engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java (hunk 5 above)
Comment (oleschoenburg) [reply engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java:352]: I've tried to explain this here: https://github.com/camunda-cloud/zeebe/pull/8802#discussion_r821897741

File: engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java (hunk 6 above)
Comment (oleschoenburg) [reply engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java:366]: I've tried to explain this here: https://github.com/camunda-cloud/zeebe/pull/8802#discussion_r821897741

File: engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbDeploymentState.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (Zelldon) [reply engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbDeploymentState.java:70]: When we implemented the distribution we added this test https://github.com/camunda-cloud/zeebe/commit/714ca6bb6cb40bb69fca809395aab67bfd8863a5#diff-c5de5f2a4beee2bb8434aa9fd91bd598357ee653f7323eb169bbf003a4c3b302 because we wanted that the complete behaves idempotent https://github.com/camunda-cloud/zeebe/commit/714ca6bb6cb40bb69fca809395aab67bfd8863a5#diff-6fd0cf018a0856491734c8c63b66cf53f54c4ee95496f79cbd4bdade2f8fcf6b

later we removed the idempotency see here https://github.com/camunda-cloud/zeebe/commit/befebf357e679833ebc3d514e40fe5f65d3cfef7#diff-6fd0cf018a0856491734c8c63b66cf53f54c4ee95496f79cbd4bdade2f8fcf6b

So i think the test can be removed

File: engine/src/test/java/io/camunda/zeebe/engine/state/instance/JobStateTest.java [hunk 13]
```diff
@@ -646,6 +646,7 @@ public void testInvariants() {
         .hasStackTraceContaining("type must not be empty");
 
     // cancel
+    jobState.create(1L, newJobRecord());
```
Comment (pihme) [inline engine/src/test/java/io/camunda/zeebe/engine/state/instance/JobStateTest.java:649]: :question: Please explain the change in this test. Is it because activate/complete now checks preconditions?

File: engine/src/test/java/io/camunda/zeebe/engine/state/migration/to_1_1/LegacyDbMessageSubscriptionState.java [hunk 14]
```diff
@@ -94,10 +94,10 @@ public void put(final long key, final MessageSubscriptionRecord record) {
 
     messageSubscription.setKey(key).setRecord(record).setCommandSentTime(0);
 
-    subscriptionColumnFamily.put(elementKeyAndMessageName, messageSubscription);
+    subscriptionColumnFamily.upsert(elementKeyAndMessageName, messageSubscription);
```
Comment (pihme) [inline engine/src/test/java/io/camunda/zeebe/engine/state/migration/to_1_1/LegacyDbMessageSubscriptionState.java:97]: :thinking: Hmm, now we get into philosophical territory. This class is a verbatim copy of the old `DbMessageSubscriptionState` anno 1.0.0. It was used to restore an old state with the old implementation, then run and test the migration.

Now the old code is changed to fit to the new API. Which is necessary and pragmatic. You also modified it in a way that is semantically the same. Philipp warned me that this day will come.

File: engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbDeploymentState.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (oleschoenburg) [reply engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbDeploymentState.java:70]: Thanks for looking into it @Zelldon, I'll remove the test then and use `deleteExisting` :+1:

**Review by pihme — CHANGES_REQUESTED** [review]
Now finished with the first pass through everything.

Hat off to the work you had with this.

Will take a closer look at the refactorings which introduced new methods, after taking a break.

For the future, it would help to have these kind of refactorings in a separate commit.

File: engine/src/main/java/io/camunda/zeebe/engine/state/message/DbMessageStartEventSubscriptionState.java [hunk 15]
```diff
@@ -64,9 +64,9 @@ public void put(final long key, final MessageStartEventSubscriptionRecord subscr
 
     messageName.wrapBuffer(subscription.getMessageNameBuffer());
     processDefinitionKey.wrapLong(subscription.getProcessDefinitionKey());
-    subscriptionsColumnFamily.put(
+    subscriptionsColumnFamily.upsert(
```
[the remark this answers is not in the document]
Comment (oleschoenburg) [reply engine/src/main/java/io/camunda/zeebe/engine/state/message/DbMessageStartEventSubscriptionState.java:67]: `shouldMergeMessageVariables` is the only test where this fails

File: engine/src/main/java/io/camunda/zeebe/engine/state/message/DbProcessMessageSubscriptionState.java [hunk 16]
```diff
@@ -70,11 +70,17 @@ public void put(final long key, final ProcessMessageSubscriptionRecord record) {
     processMessageSubscription.reset();
     processMessageSubscription.setKey(key).setRecord(record);
 
-    subscriptionColumnFamily.put(elementKeyAndMessageName, processMessageSubscription);
+    subscriptionColumnFamily.insert(elementKeyAndMessageName, processMessageSubscription);
 
     transientState.add(record);
   }
 
+  @Override
```
[the remark this answers is not in the document]
Comment (oleschoenburg) [reply engine/src/main/java/io/camunda/zeebe/engine/state/message/DbProcessMessageSubscriptionState.java:78]: I added this method to avoid `put`ing the `ProcessMessageSubscriptionRecord` again, here: https://github.com/camunda-cloud/zeebe/blob/main/engine/src/main/java/io/camunda/zeebe/engine/state/migration/DbMigrationState.java#L148-L152

This new method is in line with the other methods here, such as `updateToOpenedState`.
I guess this another change that should have been a seperate commit.

File: engine/src/main/java/io/camunda/zeebe/engine/state/message/ProcessMessageSubscription.java [hunk 17]
```diff
@@ -39,6 +39,11 @@ public boolean isOpening() {
     return stateProp.getValue() == State.STATE_OPENING;
   }
 
+  public ProcessMessageSubscription setOpening() {
```
[the remark this answers is not in the document]
Comment (oleschoenburg) [reply engine/src/main/java/io/camunda/zeebe/engine/state/message/ProcessMessageSubscription.java:42]: Added to support this change here: https://github.com/camunda-cloud/zeebe/pull/8802/files/481df8206bb182927625285a555c91f2457e957c#r822369256

File: engine/src/test/java/io/camunda/zeebe/engine/state/instance/JobStateTest.java (hunk 13 above)
Comment (oleschoenburg) [reply engine/src/test/java/io/camunda/zeebe/engine/state/instance/JobStateTest.java:649]: It's necessary so that the following cancel doesn't fail too early. In the next line we check that the cancelling of a job without type fails with a stacktrace that contains "type must not be empty". Without first creating a valid job with the same key, the cancelling instead fails earlier, because it tries to `deleteExisting` a job that doesn't exist.

File: engine/src/main/java/io/camunda/zeebe/engine/state/message/DbMessageStartEventSubscriptionState.java (hunk 15 above)
[the remark this answers is not in the document]
Comment (pihme) [reply engine/src/main/java/io/camunda/zeebe/engine/state/message/DbMessageStartEventSubscriptionState.java:67]: Ok, then leave it as is

File: engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java (hunk 4 above)
Comment (pihme) [reply engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java:220]: @korthout Could you also have a look at this and share your opinion?

Comment (korthout) [reply engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java:220]: 👍 I've had a look at the call hierarchy of `createJobRecord` and `resetVariablesAndUpdateJobRecord` and it looks good to me, but 🔧 please adjust the name of `resetVariablesAndUpdateJobRecord` (see below).

The `createJobRecord` method is only used when applying the `Job:Created` event, so an `insert` is fine here. 

The `resetVariablesAndUpdateJobRecord` is called from many places, all related to updating some property of the job (e.g. activating it, updating the number retries, failing it, timeout, etc). For these cases the `update` is expected.

I also took out my shovel and brush to do some git archeology 🕵️  and discovered that the name `resetVariablesAndUpdateJobRecord` originates from before 781f506c. Before that commit, the variables in the job record were actually modified (i.e. it had a side effect on the passed in job record). After that commit, the job record is not modified anymore. I think we should therefore change the name of this method to simply `updateJobRecord` so it mirrors the `createJobRecord` method. Next to this, we should remove the JavaDoc note on the `activate` method about resetting the variables. Of course, this is somewhat unrelated to your changes, but the naming leads to unnecessary complexity and I think we should improve it now while we have the knowledge.

File: engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbEventScopeInstanceState.java (hunk 1 above)
Comment (korthout) [reply engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbEventScopeInstanceState.java:56]: 🤔 I'm not sure, but I think we can keep it out of scope.

It's clearly unused, but the value `isAccepting` is still used by some places to check that events should be triggered. We currently have [some bugs](https://github.com/camunda-cloud/zeebe/issues?q=is%3Aissue+is%3Aopen+eventsubprocess+label%3A%22Type%3A+Bug%22) related to triggering event sub process. Perhaps this is related, but I'm not sure. Perhaps this functionality can help in resolving those bugs. I think we can keep it out of scope of these changes for now

File: engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java (hunk 4 above)
Comment (oleschoenburg) [reply engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java:214]: Thanks for looking into it @korthout :+1: I've renamed this method to `updateJobRecord` and removed the note on `activate` as suggested.

File: engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java [hunk 18]
```diff
@@ -118,22 +112,22 @@ public void activate(final long key, final JobRecord record) {
...
   }
 
   @Override
   public void recurAfterBackoff(final long key, final JobRecord record) {
     updateJob(key, record, State.ACTIVATABLE);
     jobKey.wrapLong(key);
     backoffKey.wrapLong(record.getRecurringTime());
-    backoffColumnFamily.delete(backoffJobKey);
+    backoffColumnFamily.deleteExisting(backoffJobKey);
   }
```
Comment (korthout) [inline engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java:131]: 💭 I totally agree with this change, but it made me think. What if a failed job (with backoff) is deleted before the backoff expired? I think the backoff column family entry is not deleted then. I think we have a bug there, although unrelated to this change. Might be useful to have a look or to open a bug for it. (note that this only occurs when jobs are failed with a retry which is already a niche that most users won't run into).

**Comment by korthout** [comment]
> Could you please review the changes to:
> 
> * DbElementInstanceState
> * DBJobState
> 
> And look at comments where I tagged you.
> 
> I would also appreciate if if you could be the third pair of eyes on this

I've had an extra look at `DbElementInstanceState` and `DbJobState` and the changes look all good 👍 I agree with your earlier statement "hats off to @oleschoenburg". These are not easy changes to produce. 

Same for @pihme, these are not easy changes to review 😄 Please let me know if there's anything else that you'd like to me to have an extra pair of eyes on.

File: engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java (hunk 18 above)
Comment (pihme) [reply engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java:131]: @korthout Could you create an issue for this. Even with your description I am unsure how to phrase what the problem is, tbh.

File: engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java [hunk 19]
```diff
@@ -356,7 +350,7 @@ private void createJobRecord(final long key, final JobRecord record) {
     jobsColumnFamily.insert(jobKey, jobRecordToWrite);
   }
 
-  private void resetVariablesAndUpdateJobRecord(final long key, final JobRecord updatedValue) {
+  private void updateJobRecord(final long key, final JobRecord updatedValue) {
```
Comment (pihme) [inline engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java:353]: :wrench: Maybe add JavaDoc to hint that this will not update variables

**Review by pihme — APPROVED** [review]
Long time in the making. Please merge and celebrate :tada:

**Comment by oleschoenburg** [comment]
Thanks again for the thorough review everyone. I'm really happy how this turned out :partying_face: 

bors r+

**Comment by oleschoenburg** [comment]
bors retry
